Revise cli tests

rpspringuel opened the pull request:
Here's a sample of how I'm thinking of revising the scripted cli tests.  The basic idea is to capture the expected output (either to stdout or stderr or both) and make sure it's correct.  Right now I've not expanded the number of tests, but I could see testing the stdout output and the stderr output in separate tests (e.g. stdin_and_stdin.sh would become 2 tests).

Some form of VersionManager.py (from the main repository) will probably need to be added to the repository for this to work (as version number appears in some output, e.g. version.sh).

What do people think about this revision?

davidweichiang committed:
Changes needed to work with ImageMagick 7 (tested with 7.1.1-43).
- change `convert` to `magick` and `compare` to `magick compare`
- change order of arguments to put input file before operations
- in computation of diff, remove "-background white -flatten" but I do not know why

davidweichiang committed:
Changes needed to work with ImageMagick 7 (tested with 7.1.1-43).
- change `convert` to `magick` and `compare` to `magick compare`
- change order of arguments to put input file before operations
- in computation of diff, remove "-background white -flatten" but I do not know why

davidweichiang committed:
autodetect ImageMagick

davidweichiang committed:
autodetect ImageMagick

davidweichiang committed:
left out -v somehow

davidweichiang committed:
left out -v somehow

rpspringuel committed:
Start revising cli tests

eschwab commented:
Do we need to check for gregorio version?

Or is it good enough to change:

```
EXPECTEDOUT="% File generated by gregorio 6.0.0
\GregorioTeXAPIVersion{6.0.0}%
```

to:

```
EXPECTEDOUT="% File generated by gregorio $gregorio-version
\GregorioTeXAPIVersion{$gregorio-version}%
```

`$gregorio-version` would need to be exported in `gregorio-test.sh`

davidweichiang committed:
Merge branch 'imagemagick7' of https://github.com/davidweichiang/gregorio-test into imagemagick7

rpspringuel committed:
Merge branch 'master' into cli-revisions

davidweichiang committed:
Merge branch 'ctan' into imagemagick7

rpspringuel committed:
Merge pull request #360 from davidweichiang/imagemagick7

Compatibility with ImageMagick 7

rpspringuel committed:
Merge branch 'master' into develop

rpspringuel committed:
Correct output

rpspringuel committed:
Switch to bash regex tests to make tests version agnostic

rpspringuel committed:
Split tests which produce output to both stderr and stdout

By splitting the testing of stderr and stdout output, we get more information about what is going wrong when a test fails.

rpspringuel committed:
More rewritten tests

rpspringuel committed:
Merge branch 'master' into cli-revisions

rpspringuel committed:
Revise remaining non-kpse tests

rpspringuel committed:
kpse tests rewritten

rpspringuel commented:
Okay, that should rewrite all the cli tests so that the actual output messages are checked and not the exit code.

Now that I write that, I realize that I should probably "add" tests that test the exit state.  Of course, those would simply be the original tests (or at least equivalent to them).

rpspringuel committed:
Exit state tests

While one could argue that the original tests were based really tests of the exit state, I decided not to rewrite the tests in a way that makes explicit that we are testing the exit state and what we expect the exit state to be.  Since we only use exit states of 0 (success) and 1 (failure) that may seem like overkill, but this leaves open the possibility of us making a more expansive use of exit states in the future.

rpspringuel committed:
Replace gregorio with $gregorio

The error messages for a few tests use the actual name of the invocation rather than just "gregorio" in a few more places than I originally thought.

rpspringuel commented:
Can anyone explain the difference between the two kpsewhich tests? `bad-response.sh` and `many-vowel.sh` seem to test exactly the same thing as the exported environmental variable doesn't appear to make a difference on my system.

rpspringuel committed:
Refining scripted tests in other directories

Still need to do the kpsewhich tests, but I need to understand the difference between them first.

rpspringuel committed:
Merge commit '0a5e0c7a4ddf2c60d85fa4d2e12ac19469907e59' into develop

rpspringuel committed:
Merge branch 'ctan' into develop

rpspringuel committed:
Merge branch 'ctan' into develop

rpspringuel committed:
Merge branch 'develop' into cli-revisions

rpspringuel committed:
Merge branch 'ctan' into develop

rpspringuel committed:
Merge branch 'develop' into cli-revisions

rpspringuel commented:
Okay, I've figured out what `bad-response.sh` and `many-vowel.sh` are supposed to be doing, but I'm not sure they are actually doing it.  There's a local `kpsewhich` script in the test that should return either nothing (bad-response) or the list of dat files in the test folder (many-vowel) instead of what the actual `kpsewhich` would return (the locations of the various `gregorio-vowels.dat` files on your system).  However, in my testing, it appears that the local script is not used when conducting the test (despite the alteration to `PATH` that's supposed to ensure that it is).  Gregorio seems to be finding the real `kpsewhich` and using that.  (Adding the `-v` option to the call gets gregorio to print out the vowel files it consulted, and this list corresponds to the ones the actual `kpsewhich` finds, not the ones our pseudo-`kpsewhich` does.)

I've double checked that I'm using the gregorio that comes with TeX Live (i.e. one which is compiled with the kpathsea libraries).  This may be revealing a problem with our use of the kpathsea libraries.  Can anyone confirm that?
